packages/server/example/monaco_editor/README.md

```diff
@@ -0,0 +1,10 @@
+## SQLLanguageServer monaco-editor example
+
+### how to setup
+
+```sh
+$ yarn && yarn build
```

Comment:
This doesn't seem to work for me.
```
git/sql-language-server/packages/server/example/monaco_editor $ yarn build
yarn run v1.22.4
$ yarn build:server && yarn build:client
$ cd ../../ && yarn
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/5] Validating package.json...
warning sql-language-server@0.10.7: The engine "vscode" appears to be invalid.
[2/5] Resolving packages...
success Already up-to-date.
$ yarn compile:sqlint && yarn compile:cli && yarn compile:index
$ cd ../sqlint && yarn build
$ rm -rf dist && rollup -c
internal/modules/cjs/loader.js:960
  throw err;
  ^

Error: Cannot find module 'git\sql-language-server\node_modules\rollup\dist\bin\rollup'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)
    at Function.Module._load (internal/modules/cjs/loader.js:840:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
Umm it works for me even if I get this branch from scratch..it's weird.
Could you try to build sqlint package manually by following steps?

```sh
$ cd packages/sqlint/
$ yarn && yarn build
```

Reply:
@joe-re No good.
```
git\sql-language-server\packages>yarn && yarn build
yarn install v1.22.4
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/5] Validating package.json...
warning sql-language-server@0.10.7: The engine "vscode" appears to be invalid.
[2/5] Resolving packages...
success Already up-to-date.
$ npm i sqlite3 electron-rebuild
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

> sqlite3@4.2.0 install git\sql-language-server\node_modules\sqlite3
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using request for node-pre-gyp https download
[sqlite3] Success: "git\sql-language-server\node_modules\sqlite3\lib\binding\node-v72-win32-x64\node_sqlite3.node" is installed via remote
+ electron-rebuild@1.11.0
+ sqlite3@4.2.0
removed 32 packages, updated 2 packages and audited 229 packages in 6.182s

12 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

Done in 7.77s.
yarn run v1.22.4
error Command "build" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
The only change I have on my computer is `package-lock.json`. Perhaps you can try checking out in a completely new folder and redoing the steps and then committing your `package-lock.json` file?
```
git/sql-language-server $ git branch
  master
* monaco-editor-example
```
```
git/sql-language-server $ git status
On branch monaco-editor-example
Your branch is up to date with 'origin/monaco-editor-example'.  

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        package-lock.json

nothing added to commit but untracked files present (use "git add" to track)
```

Reply:
Whoops, wrong folder. Here it is in the `packages/sqlint` folder.
```
$ yarn && yarn build
yarn install v1.22.4
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.    
[1/5] Validating package.json...
warning sql-language-server@0.10.7: The engine "vscode" appears to be invalid.
[2/5] Resolving packages...
success Already up-to-date.
$ yarn run build
yarn run v1.22.4
$ rm -rf dist && rollup -c
internal/modules/cjs/loader.js:960
  throw err;
  ^

Error: Cannot find module 'git\sql-language-server\node_modules\rollup\dist\bin\rollup'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15)
    at Function.Module._load (internal/modules/cjs/loader.js:840:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```
I should also point out that `rm -rf` won't work on Windows if someone is using the Command Prompt.

Reply:
> Perhaps you can try checking out in a completely new folder and redoing the steps and then committing your package-lock.json file?

Actually I did it and there were no errors on my env..
Did you do something npm command? Basically this project's using yarn and depending on `yarn workspace`. So npm doesn't work for this and that's the case `package-lock.json` should be not generated.

> I should also point out that rm -rf won't work on Windows if someone is using the Command Prompt.
I see. Thank you for pointing it out. Maybe you're using windows machine?
I don't have a windows environment right now and haven't tried to build it on windows.
Maybe the above error has also happened on only windows..?
I succeeded to build it on both Ubuntu18 and MacOS.

Reply:
> Maybe the above error has also happened on only windows..?
> I succeeded to build it on both Ubuntu18 and MacOS.

@joe-re It does build on macOS for me but there are a bunch of errors in my console. I also don't know what I should do to trigger a quick fix.
```
$ yarn start
yarn run v1.19.2
$ node lib/server.js
using faster connection
ready to launch server
TypeError: diagnostics.concat(...).flat is not a function
    at rules.forEach (github/sql-language-server/packages/server/dist_index/index.js:48994:60)
    at Array.forEach (<anonymous>)
    at apply (github/sql-language-server/packages/server/dist_index/index.js:48972:11)
    at walk (github/sql-language-server/packages/server/dist_index/index.js:49003:38)
    at Object.values.forEach (github/sql-language-server/packages/server/dist_index/index.js:49005:23)
    at Array.forEach (<anonymous>)
    at walk (github/sql-language-server/packages/server/dist_index/index.js:49004:25)
    at execute (github/sql-language-server/packages/server/dist_index/index.js:48959:12)
    at lint (github/sql-language-server/packages/server/dist_index/index.js:60098:45)
    at doLint (github/sql-language-server/packages/server/dist_index/index.js:60160:31)
Error: No files matching 'undefined' were found.
    at lint (github/sql-language-server/packages/server/dist_index/index.js:60094:15)
    at doLint (github/sql-language-server/packages/server/dist_index/index.js:60160:31)
    at createDiagnostics (github/sql-language-server/packages/server/dist_index/index.js:60186:23)
    at documents.onDidChangeContent (github/sql-language-server/packages/server/dist_index/index.js:111527:29)
    at CallbackList.invoke (github/sql-language-server/packages/server/dist_index/index.js:409:39)
    at Emitter.fire (github/sql-language-server/packages/server/dist_index/index.js:468:36)
    at connection.onDidChangeTextDocument (github/sql-language-server/packages/server/dist_index/index.js:6282:38)
    at handleNotification (github/sql-language-server/node_modules/vscode-jsonrpc/lib/main.js:502:43)
    at processMessageQueue (github/sql-language-server/node_modules/vscode-jsonrpc/lib/main.js:273:17)
    at Immediate.setImmediate [as _onImmediate] (github/sql-language-server/node_modules/vscode-jsonrpc/lib/main.js:260:13)
ready to launch server
TypeError: diagnostics.concat(...).flat is not a function
    at rules.forEach (github/sql-language-server/packages/server/dist_index/index.js:48994:60)
    at Array.forEach (<anonymous>)
    at apply (github/sql-language-server/packages/server/dist_index/index.js:48972:11)
    at walk (github/sql-language-server/packages/server/dist_index/index.js:49003:38)
    at Object.values.forEach (github/sql-language-server/packages/server/dist_index/index.js:49005:23)
    at Array.forEach (<anonymous>)
    at walk (github/sql-language-server/packages/server/dist_index/index.js:49004:25)
    at execute (github/sql-language-server/packages/server/dist_index/index.js:48959:12)
    at lint (github/sql-language-server/packages/server/dist_index/index.js:60098:45)
    at doLint (github/sql-language-server/packages/server/dist_index/index.js:60160:31)
```

Reply:
Ah sorry, it's using Array.flat and needs node of version 11 or more.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

Reply:
> It does build on macOS for me but there are a bunch of errors in my console

Sorry to bother you.. I'll try it on some WIndows env too somehow.

Reply:
> Ah sorry, it's using Array.flat and needs node of version 11 or more.

@joe-re You're right, I'm on Node.js v10.16.3. I will upgrade Node.js with `brew`. Could you please update your documentation to reflect this requirement? Thanks!

Reply:
> Could you please update your documentation to reflect this requirement? Thanks!

Of course! Your points are very helpful to improve it. Thank you so much!

Reply:
@joe-re I'm able to reproduce the error now but also see something (presumably unrelated) to the error in my console.
```
/usr/local/Cellar/node/14.5.0/bin/node $ lib/server.js
using faster connection
ready to launch server
Error: No files matching 'undefined' were found.
    at lint (github/sql-language-server/packages/server/dist_index/index.js:60094:15)
    at doLint (github/sql-language-server/packages/server/dist_index/index.js:60160:31)
    at createDiagnostics (github/sql-language-server/packages/server/dist_index/index.js:60186:23)
    at github/sql-language-server/packages/server/dist_index/index.js:111527:29
    at CallbackList.invoke (github/sql-language-server/packages/server/dist_index/index.js:409:39)
    at Emitter.fire (github/sql-language-server/packages/server/dist_index/index.js:468:36)
    at github/sql-language-server/packages/server/dist_index/index.js:6267:38
    at handleNotification (github/sql-language-server/node_modules/vscode-jsonrpc/lib/main.js:502:43)
    at processMessageQueue (github/sql-language-server/node_modules/vscode-jsonrpc/lib/main.js:273:17)
    at Immediate._onImmediate (github/sql-language-server/node_modules/vscode-jsonrpc/lib/main.js:260:13)
ready to launch server
```

Reply:
You don't need to concern about that error. It doesn't affect anything for working. It's already fixed on master branch.
https://github.com/joe-re/sql-language-server/pull/38